extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -1197,8 +1198,18 @@ private DataSegmentTimelineView makeDataSegmentTimelineView()
       }
 
       // Fetch all published, used segments (all non-realtime segments) from the metadata store.
-      final Collection<DataSegment> publishedUsedSegments =
-          FutureUtils.getUnchecked(context.coordinatorClient().fetchUsedSegments(dataSource, intervals), true);
+      // If the task is operating with a REPLACE lock,
+      // any segment created after the lock was acquired for its interval will not be considered.
+      final Collection<DataSegment> publishedUsedSegments = new HashSet<>();
+      for (Interval interval : intervals) {
```

Comment:
Looking at `RetrieveSegmentsToReplaceAction` and the method `IndexerMetadataStorageConnector.retrieveUsedSegmentsAndCreatedDates`, I suppose we could just pass a list of intervals to the action rather than firing a separate action for each interval.